### PR TITLE
Per-version inspection toggle

### LINKED ISSUE #102: Disable inspections if the ZIO version is less than the one in which the corresponding feature was introduced
Needs investigation but this is probably possible. However, one should be careful with version comparison. It might be tricky 🙂

### PR DESCRIPTION
**Pull request by hmemcpy**
Fixes #102 

This solves the immediate problem by allowing to specify `availableSince` version, however this adds an extra minute to the test run, meaning this could affect performance. Need to investigate caching.

### CODE REVIEW

File: src/main/scala/zio/intellij/utils/package.scala [hunk 1]
```diff
@@ -122,4 +115,25 @@ package object utils {
...
+      (for {
+        library <- module.libraries
+        url     <- library.getUrls(OrderRootType.CLASSES)
+        if p(url)
+        trimmedUrl  = utils.trimAfterSuffix(url, ".jar")
+        versionStr <- LibraryExt.runtimeVersion(trimmedUrl)
+        version    <- Version.parse(versionStr)
+      } yield version).headOption
+
+    def zioVersion: Option[Version] = findLibrary(lib => lib.contains("/dev/zio/zio_") || lib.contains("/dev.zio/zio_"))
```
Comment (hmemcpy) [inline src/main/scala/zio/intellij/utils/package.scala:133]: For some reason, the test Ivy downloader decided to use the `dev.zio` directory structure 🤷‍♂️

**Comment by hmemcpy** [comment]
This PR enables all inspections for ZIO versions >= 1.0.0.
It might stop working for people who are still on the pre-1.0 RC...

**Comment by timaliberdov** [comment]
LGTM. Should we review every inspection and override the `isAvailable` method for things that came after 1.0.0?

**Review by timaliberdov — APPROVED** [review]

**Comment by hmemcpy** [comment]
We should, yeah. Although I just realized this is only for Inspections. This won't work for the reason you needed it - accessible stuff....

I need those changes for the testrunner PR I'm working on.

**Pull request merged**